Release 9.3.1

clairesunstudio opened the pull request:
## 9.4.0 (03/28/2019)

### Added
- (Docs) Add an issue template for submitting Mayflower project to the implementation listing #520

### Changed
- (React) [*] DP-12932: Upgrades storybook and related add-ons to version 5.0.x. #516
- (React) [*] DP-12932: Adds storysource and a11y add-ons. #516

isaacchansky commented:
Lots of awesome documentation, nice! 

The twig changes change the data model and imply a major version update due to breaking changes. Where are we documenting the changes that may need to occur?

legostud commented:
@isaacchansky - I see a couple of CSS files that I forgot to leave the old selectors in (corrected now), but all the changes I made were supposed to be Minor changes.  Which changes did you feel fell within a Major release?

* Button.twig - enhancement to add an option for rendering as a link.  I also modified the CSS classes in the Twig, but left the old selectors in the CSS file to work with an older Twig file
* Figure.twig - Changed the CSS class for alignment, but left the old CSS selectors
* Video.twig - Bug Fix -> Syntax change for Twig logic
* Callout Link - Bug fix -> using title attr vs span tag - CSS is still there to hide the span if the html doesn't get updated
Download Link - Bug fix -> Syntax change for Twig Logic
Floating Action - Change the CSS class for alignment, but left the old CSS selectors
Social Links - Bug fix -> using title attr vs span tag
Action Finder - Bug fix -> incorrect logic to hide the filter input

isaacchansky commented:
Ah I understand, I was misreading some `button.twig` stuff. I appreciate the clarification, this looks solid.

jesconstantine commented on styleguide/source/_patterns/01-atoms/01-buttons/button.md:
```diff
@@ -1,26 +1,32 @@
+* Using a [link](./?p=atoms-button-as-link) instead of a button
+* [small](./?p=atoms-button-as-small)
+* [outline](./?p=atoms-button-as-outline)
+* [gray](./?p=atoms-button-as-alt-color) color
```
Should this be `button-as-alt-color` or `button-as-gray`?  I actually kind of like the non-color variable names so that if we ever decided to change colors schemes, we don't have to rename patterns.  What do you think?

legostud replied:
I fixed the links in a later PR so I'll cherry pick them into this PR.  

We can't really use 'alt' anymore since we have the green and gray alternatives.  I could write them as --secondary (green) and --tertiary (gray)

legostud replied:
I went with 'quaternary' for gray, because that match the color themes for the site. 'tertiary' is current yellow so I wrote the CSS to support this, but didn't add it as a variant.

jesconstantine commented on styleguide/source/_patterns/01-atoms/01-buttons/button.md:
```diff
@@ -1,26 +1,32 @@
+* [small](./?p=atoms-button-as-small)
+* [outline](./?p=atoms-button-as-outline)
+* [gray](./?p=atoms-button-as-alt-color) color
+* [green with an outline](./?p=atoms-button-as-alt-color)
```
Same comment as above: Should this be `button-as-alt-color` or `button-as-green`? (the link doesn't actually work for me locally)  I actually kind of like the non-color variable names so that if we ever decided to change colors schemes, we don't have to rename patterns.  What do you think?

jesconstantine commented on styleguide/source/_patterns/01-atoms/01-buttons/button.twig:
```diff
@@ -1,12 +1,19 @@
-{% if button.theme %}
-  {% set buttonClass = buttonClass ~ "-" ~ button.theme %}
-{% endif %}
+{# Depreciated after version 5.0 #}
```
nice 👍

jesconstantine commented on styleguide/source/_patterns/01-atoms/01-buttons/button.twig:
```diff
@@ -1,12 +1,19 @@
+  <a 
+    href="{{ button.href }}" 
+    class="ma__button {{ buttonSize }} {{buttonColor}} {{buttonStyle}}"
+    title="{{ button.info }}">
```
Curious as to the motivation behind using the title attribute here vs a visually hidden span?

legostud replied:
Title is designed for this purpose so why not use it over doing a hack with a hidden span tag.  Also using the attribute allows you to pass a string that doesn't have to be an extension of the button's text.

"view more" + "about this thing" can instead be "click here to read more about this thing"

jesconstantine commented on styleguide/source/_patterns/01-atoms/04-headings/comp-heading.json:
```diff
@@ -1,6 +1,6 @@
+    "title": "Title text",
     "sub": "",
     "color": "",
     "id": "employment",
```
Is it worth updating the `id` to `title-text`?  I like that it would show a potential format.  Is it also worth mentioning in the docs for this property that the string should contain those characters accepted for an id attribute: ASCII letters, digits, '_', '-' and '.', or maybe that's self explanatory?

legostud replied:
The id value doesn't have to match the title value.  Id could be GUID89173489 instead which would actually be better.

jesconstantine replied:
I'd still be in favor of updating the id to something like GUID9827924.

jesconstantine commented on styleguide/source/_patterns/01-atoms/04-headings/comp-heading.md:
```diff
@@ -1,38 +1,26 @@
+    type: string ("yellow") / optional
   id:
-    type: string
+    type: string (unique per page) / optional
```
unique per page - 👍

jesconstantine commented on styleguide/source/_patterns/01-atoms/09-media/video.md:
```diff
@@ -1,14 +1,32 @@
+  link: 
+    type: decorativeLink / optional
+  position: 
+    type: string (null, 'right') / optional
```
In other places I've seen `''` as opposed to `null` - can you explain when it's appropriate to use which, for my own learning?

jesconstantine replied:
Also curious why left isn't an option -- is that just based on our design language?

legostud replied:
--left would make sense as a variant, but that option has not been added to Mayflower (css doesn't exist)

I think I was inconsistent here.  '' and null are pretty interchangeable so I haven't really thought about it much.  I think if the variable is Type String then we should use '' and if it's Type Object we should use null.  Does that sound good?

jesconstantine replied:
re: `--left` I'll make a note on the pattern to consider it as an update

re: '' vs null -- yes that makes sense to me!

jesconstantine commented on styleguide/source/_patterns/01-atoms/10-table/table.md:
```diff
@@ -1,52 +1,43 @@
----
-Description: an atom for `<table>` elements with child `<thead>`, `<tbody>`, `<tr>`, and `<td>` elements.
+### Description
+This Displays a styled table used in a Rich Text Editor
```
Can we add a note to see `[@organisms/tabular-data](/?p=organisms-tabular-data)` pattern for data table needs?

legostud replied:
Not sure I want to link an Atom to an Organism.  Tabular Data should be linking to the Atom instead.  I think I fixed this in a later PR, let me check.

legostud replied:
What are the "Data Table needs"?

jesconstantine replied:
Disregard.  :)

jesconstantine commented on styleguide/source/_patterns/01-atoms/decorative-link.md:
```diff
@@ -1,34 +1,21 @@
+    type: string / required
+  info: 
+    type: string (adds more description to the link) / optional
+  property:
```
Is this a schema.org legacy property, can it be removed?

legostud replied:
yep.  Thought I had already.

jesconstantine commented on styleguide/source/_patterns/01-atoms/decorative-link.md:
```diff
@@ -1,34 +1,21 @@
+    type: string (url) / required
+  text:
+    type: string / required
+  info: 
```
Curious about using `aria-label` for this property in the twig template here vs a visually hidden span, or the `title` attribute used in the button pattern `a` element?  I also think it might be nice to talk about using this for more context for links and illustrating this in the pattern it self (as in direction links for contacts).

legostud replied:
I'll switch this to a title attribute.  I agree we need a better discussion about when to use 'title', 'aria-label', or a hidden 'span' tag.

jesconstantine replied:
Let's try to have that discussion this cycle and make sure we capture the outcome in documentation.

legostud replied:
That first link seems to support my decision to use the 'title' attribute since it works across all screen readers (VO on OSX requires manually triggering it, but it will read it).

The second link is from 2012 and browsers have evolved over the past five years so I'm not sure how relevant the content is now.

jesconstantine commented on styleguide/source/_patterns/02-molecules/callout-link.json:
```diff
@@ -1,8 +1,7 @@
-    "type": "",
     "href": "#",
-    "info": ""
+    "info": "Order a MassParks Pass online through Reserve America"
```
See [earlier comment](https://github.com/massgov/mayflower/pull/521/files#r131136658) about providing an insightful example for this property with links.  What do you think?

jesconstantine commented on styleguide/source/_patterns/02-molecules/callout-link.twig:
```diff
@@ -1,7 +1,7 @@
-<a class="ma__callout-link" href="{{ calloutLink.href }}">
+<a class="ma__callout-link" href="{{ calloutLink.href }}" title="{{ calloutLink.info }}">
```
@isaacchansky Do you want to track your note about the `callout-link` vs `illustrated-link`here?

legostud replied:
It will eventually become a mute point since the Illustrated Link is slated to be depreciated as part of the updates to the Callout Link.

jesconstantine commented on styleguide/source/_patterns/02-molecules/callout-link.md:
```diff
@@ -1,25 +1,20 @@
-    type: string/required
-  type:
-    type: string/optional ("external")
+    type: string / required
```
is this only either "internal" or "external"?  Should we note that here?

legostud replied:
'internal' isn't required since that's the default.  I should probably say ('', 'external')

jesconstantine replied:
Oh sorry, missed that.  `type` is removed altogether.  Got it.

jesconstantine commented on styleguide/source/_patterns/02-molecules/download-link.json:
```diff
@@ -3,7 +3,7 @@
+      "text": "Link to a file for download",
       "href": "#",
       "info": "",
       "property": ""
```
This can be removed, if it isn't already in a different PR.

jesconstantine commented on styleguide/source/_patterns/02-molecules/icon-links.md:
```diff
@@ -1,22 +1,20 @@
+### Status
+* Stable as of 5.0.0
+
+### Pattern Contains
```
I do see this show up in lineage, are we adding it manually for all patterns as well?  If so, can we link to the pattern?

legostud replied:
I'm manually adding it here for quicker reference.  Personally, I'd rather not make them links since that would be hard to keep up to date and we already have the link in Lineage.  When we start organizing Patterns into proper folders, we would need to also modify the documentation for any Pattern that contains it, whereas the Lineage links will just automatically update.

  Which would become  especially since one of the tasks for this Sprint it's already linked in the Lineage,

jesconstantine commented on styleguide/source/_patterns/02-molecules/main-nav.md:
```diff
@@ -1,9 +1,13 @@
-## State: BETA
-### Required variables
+### Description
+The main navigation for the site.
```
Is there a hard limit on being 1 level deep?  Is that worth adding here?  I saw (and loved!) for the list patterns you mentioned you could include an optional nested sublist, so I'm thinking along those lines here.

legostud replied:
We haven't designed the Main Nav to have an optional sublist.  It is restricted to one level deep.

jesconstantine replied:
Right I think it's worth mentioning that that's what the pattern is designed to support.  I know for Mass.gov people know that but I don't think it's crystal clear out of Mass.gov context.

jesconstantine commented on styleguide/source/_patterns/03-organisms/by-author/action-finder.md:
```diff
@@ -1,73 +1,46 @@
---- 
-title: Action Finder
```
I think this is a pattern we should rename (I'll add a note to the spreadsheet). What do you think?

legostud replied:
debatable.  I've heard it call the Service Finder, but it's also used on a Service page so obviously it's not always a Service Finder.  I'm okay with renaming it, I'm just not sure what to call it.

I also think it should be refactored to use a new Molecule which is a grid of Callouts and Illustrated Links.  We can do the same for the Key Actions Pattern.

jesconstantine replied:
re: the name I don't think it should reflect the name of content at all - so neither action nor service.  It's really a filterable card grid right?  That's more along the lines of what I was thinking.  What do you think?

re: refactoring to use the grid molecules - yes!

jesconstantine commented on styleguide/source/_patterns/03-organisms/by-author/action-finder.md:
```diff
@@ -1,73 +1,46 @@
+~~~
+actionFinder: {
+  id:
+    type: string/uniqueID/required 
```
I like this as a way to describe ids. 👍

jesconstantine commented on styleguide/source/_patterns/03-organisms/by-author/action-finder.twig:
```diff
@@ -5,7 +5,7 @@
 {% endif %}
 
-{% if actionFinder.links|length <= 6 and hideFilter %}
+{% if actionFinder.links|length <= 6 or hideFilter %}
```
Is 6 the number determined from a us / design perspective?  I wonder if it shouldn't be configurable and / or if the pattern shouldn't just look for the `hideFilter` flag and if the implementers / consumers should use business logic to determine when to set it.  Thoughts?

legostud replied:
It was part of the original design Language.  In reality, the designs no longer use the filter so we could just remove it.

legostud replied:
Looks like another team has now removed it as part of a bug fix.

jesconstantine commented on styleguide/source/_patterns/03-organisms/by-author/form-downloads.md:
```diff
@@ -0,0 +1,27 @@
+### Description
+This pattern shows a list of links to downloadable files and online forms.  You can also include a link to another page, but that should be used sparingly.
```
When is a good use case for  including a link to another page?

legostud replied:
Beats me, but it was part of the Design for one of the later pages even though it doesn't really make since to be part of a list of downloadable files.

jesconstantine replied:
Okay.  I'd like to add something eventually but that's more around usage guidelines.

jesconstantine commented on styleguide/source/_patterns/03-organisms/by-author/key-actions.md:
```diff
@@ -0,0 +1,30 @@
+### Description
+This pattern shows a grid containing a combination of Callout links and Illustrated Links beneath an optional title
```
This is another tricky one to me because it feels awfully similar to the action finder.  If it's not feasible to make consolidate them, I think it's another good candidate to rename.

legostud replied:
It's different enough that I don't think they should be consolidated.  I do think this could be included as part of the Action Finder though with the title turned off.

jesconstantine replied:
I think it's another candidate for renaming.  The concept of "action" is massgov specific.  I'm thinking something like card link grid...

jesconstantine commented on styleguide/source/_patterns/03-organisms/by-author/link-list.md:
```diff
@@ -1,24 +1,26 @@
 
-### Notes
+### Pattern Contains
+* Comp Heading atom (optional)
```
You haven't been including the pattern type in "contains", I'm guessing this is a typo?

legostud replied:
yep

jesconstantine commented on styleguide/source/_patterns/03-organisms/by-author/link-list.md:
```diff
@@ -36,5 +38,4 @@ linkList : {
     type: array of object (decorative-link) / required
```
The type for the child patterns in the "Variables" section here seems inconsistent with how it's been happening on other patterns.

jesconstantine commented on styleguide/source/_patterns/03-organisms/by-author/link-list.md:
```diff
@@ -1,24 +1,26 @@
 
+### Variant options
+* See Variants for Comp Heading
+* [Short List](./?p=organisms-link-list-with-short-list) by passing less than four items
```
FYI this link was rendering the default variant for me... could be a local issue though because the file names all seemed to match and the data json only had 3 links...

jesconstantine replied:
I see similar behavior for split columns and split columns 2 -- and now it's reminding me: this is the bug with the data inheritance / lineage, right?  So we should test these patterns with #519 ?

legostud replied:
yep, that's the issue with the Pattern Lab upgrade.  There is a pull request to fix that if you'd like to review it.

jesconstantine commented on styleguide/source/_patterns/03-organisms/by-author/mapped-locations.md:
```diff
@@ -1,27 +1,33 @@
 
-### Required Variables
+### Usage Guidelines
+* The decorative link is placed above the map for accessibility reasons.
```
Do you think we should add something like "to link to a more accessible list of locations"

legostud replied:
hmm, the comment I made was about the placement of the link itself since normally the link would go after the Map like we do for the Video pattern and other view more links

jesconstantine commented on styleguide/source/_patterns/03-organisms/by-author/rich-text.md:
```diff
@@ -0,0 +1,43 @@
+    }
+  }]
+}
+~~~
```
Love the docs for this pattern!

jesconstantine commented on styleguide/source/_patterns/03-organisms/by-template/page-banner.md:
```diff
@@ -1,25 +1,22 @@
     type: string/required (path to image for narrow screens)
   size:
-    type: string(optional) ('small', 'large', 'overlay')
+    type: string(optional) (null, 'small', 'medium', 'large', 'overlay', 'columns')
```
`''` vs `null` per our earlier conversation

jesconstantine commented on styleguide/source/_patterns/04-templates/01-content-types/services.md:
```diff
@@ -0,0 +1,87 @@
+### Description
+Displays a collect of components to help describe an available Service on Mass Gov
```
typo "collection"

jesconstantine commented on styleguide/source/_patterns/04-templates/01-content-types/services.md:
```diff
@@ -0,0 +1,87 @@
+
+
+### Variant options
+* Most of the elements on the page are optional.  The Page Banner, Intro text, and Google Map are required on this template.
```
Nice 👍

legostud commented:
@jesconstantine - I think I've updated all of the issues or provided comments.

jesconstantine commented on styleguide/source/_patterns/01-atoms/01-buttons/button.md:
```diff
@@ -1,26 +1,32 @@
     type: string / optional ("" or "small")
   theme:
-    type: string / optional ("" or "alt")
+    type: string / optional ("", "secondary", or "secondary")
```
I'm guessing one of these "secondary"s should be "quaternary", right?

legostud replied:
yep, good catch

clairesunstudio committed:
Merge pull request #514 from massgov/master

Master

clairesunstudio committed:
GitBook: [develop] 25 pages and 107 assets modified

clairesunstudio committed:
Merge pull request #515 from massgov/master

Sync Master into Develop

smurrayatwork committed:
React/DP-12932:  Storybook v5 upgrade. (#516)

* Upgrades storybook to version 5.0.x, adds accessibility, storysource addons.

* Updates backstop, fixes backstop configuration for storybook 5 paths.

* Updates to 5.0.5

* Adds changelog.

* update button colour for add-info

* Fixes decorator for withInfo to be global.

* remove with info decorator from individual stories

* bug fix

* bug fix

* move storybook with info bottom to bottom right

* update reference images

* Update changelogs/DP-12932.md

* Update changelogs/DP-12932.md

* testing delay

* test again

* update circle node version

* update package json

* test webpack cahnge temporarily

* Revert "test webpack cahnge temporarily"

This reverts commit 7710db4973dbe1474e7368d18e78c5bccd86ab8b.

* test reverting webpack config

* Revert "test reverting webpack config"

This reverts commit 89734d893dcc77a452c0855660fd727c950431e2.

* Reverts backstopjs version.

* Update react/backstop/backstop.js

clairesunstudio committed:
New issue template

clairesunstudio committed:
Add an issue template (#520)

* Add an issue template

Add an issue template for submitting Mayflower project to the implementation listing

* update url

clairesunstudio committed:
add changelog
